src/fuse_pointwise.cpp

```diff
@@ -45,7 +45,14 @@ static literal get_scalar(instruction_ref ins)
         return {};
     auto e = ins->eval();
     literal r{};
-    e.visit_at([&](auto x) { r = literal{x}; });
+    if(e.get_shape().type() == shape::bool_type)
+    {
+        r = literal{e.at<bool>()};
```

Comment:
Just add comment here that during `visit` bool type are treated as `int8` and that's why this requires `if` condition to handle this case separately.

Reply:
will do. I'll rebase this off develop too.

Reply:
done